src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -424,6 +424,14 @@ def _node_from_int(var):
 
         self._datatype = datatype
 
+    @property
+    def datatype(self):
+        '''
+        :returns: the datatype of each element in the array.
+        :rtype: :py:class:`psyclone.psyir.symbols.DataSymbol`
```

Comment:
I see that this clashes with #1857. I checked where I need this - it's to verify if a symbol already exists in the symbol table that it is the right type (line 188 in LFRicSymbolTable), and I think this consistency check is important to make sure we have the right declaration in place (it certainly was very useful for me while working on this).

So my suggestion is to leave this for now, and when #1857, it will do it properly. I looked first at self._intrinsic, but its content is inconsistent, which I think is what #1857 is about. The usage is triggered by any kind of arrays.

Reply:
OK.